components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -154,8 +154,15 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             String retryParam = "";
 
             if (context.isRetrying()) {
-                retryParam = BasicAuthenticatorConstants.AUTH_FAILURE_PARAM + "true" +
-                        BasicAuthenticatorConstants.AUTH_FAILURE_MSG_PARAM + "login.fail.message";
+                if (context.getProperty("InvalidEmailUsername") != null &&
+                        (Boolean) context.getProperty("InvalidEmailUsername")) {
```

Comment:
shall we define a constant for InvalidEmailUsername

Reply:
also we can use Boolean.parseBoolean(context.getProperty("InvalidEmailUsername") which will handle the null check as well